### PR TITLE
Weather app page 255

### PR DESCRIPTION
**Pull request by ran-helfer**

### CODE REVIEW

File: RxSwiftTest/WheatherAPI.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    func observableRequestForModel<T: Decodable>(request: URLRequest, emptyType: T) -> Observable<T> {
+        return URLSession.shared.rx.data(request: request).map({ data in
+            do {
+                let model = try JSONDecoder().decode(T.self, from: data)
+                return model
+            } catch let error {
+                return emptyType
+            }
+        })
+    }
```
Comment (ran-helfer) [inline RxSwiftTest/WheatherAPI.swift]: Dov, Elaz, i think we can use something like this in our app ? just need some polish and removal of `emptyType`

What do you think ?

Comment (Elaz-viz) [reply RxSwiftTest/WheatherAPI.swift]: we have similar methods in ApolloClient+Reactive, using `Obsrvable.create ...`, which is even easier for creating observables - but I think we should discuss if new apis should still use rxSwift or move to combine, especially considering SwiftUI's extensive use of that.

Comment (ran-helfer) [reply RxSwiftTest/WheatherAPI.swift]: Thanks !  Sounds like a good point

Just started with `RxSwift` and it is no longer relevant... 🐰

Comment (Elaz-viz) [reply RxSwiftTest/WheatherAPI.swift]: Combine is very similar, it's the exact same concepts, just with some added flavours.

**Pull request closed without merging**